app/battle.py

```diff
@@ -0,0 +1,48 @@
+class Battle:
+    def __init__(self, profile):
+        self.profile = profile
+
+    def armour_preparation(self):
+        for stat in self.profile.values():
+            stat["protection"] = 0
+            for a in stat["armour"]:
+                stat["protection"] += a["protection"]
+        return self.profile

```

Comment:
I think it's better to pass required elements to the function and return them, than work with entire set of object atributes.